### DISCUSSION TITLE
List index out of range in BAC0.whois()

### DISCUSSION THREAD

**Discussion by devsurya1605** [ChristianTremblay/BAC0#453] [Q&A]
I'm encountering an IndexError: list index out of range when using BACnet.whois() in my Jupyter notebooks for BACnet communication. I have two separate notebooks:

1. Simulator: This notebook establishes a connection to a BACnet simulator using BAC0.connect().
2. Reader: This notebook attempts to discover BACnet devices on the network using BACnet.whois()
Both notebooks are running on the same machine, and the simulator is configured with the IP address '192.168.1.48'. However, the Reader notebook throws the IndexError.

```
from BAC0 import connect  # Assuming this is my import statement

BACnet = BAC0.connect()
mac, device_id = BACnet.whois()[0]  # This line throws the error

```

I've verified that the BACnet simulator is running on the correct network segment.  What could be causing this error? Are there potential conflicts between how the connect() function is used in the simulator and the whois() function in the reader?

**Comment by ChristianTremblay** [comment]
try
```
import BAC0
bacnet = BAC0.lite()

res = bacnet.whois()
```

then play with the results

**Comment by devsurya1605** [comment]
Thanks Christian
